include/zephyr/bluetooth/mesh/cfg_cli.h

```diff
@@ -812,9 +836,9 @@ struct bt_mesh_cfg_mod_pub {
  *
  *  @return 0 on success, or (negative) error code on failure.
  */
-int bt_mesh_cfg_mod_pub_get(uint16_t net_idx, uint16_t addr, uint16_t elem_addr,
-			    uint16_t mod_id, struct bt_mesh_cfg_mod_pub *pub,
-			    uint8_t *status);
+__deprecated int bt_mesh_cfg_mod_pub_get(uint16_t net_idx, uint16_t addr, uint16_t elem_addr,
+					 uint16_t mod_id, struct bt_mesh_cfg_cli_mod_pub *pub,
```

Comment:
This will make users have to change the definition in their program, when using the deprecated functions. It should be the deprecated definition in the deprecated functions, and typecast in the function calls to the \*cli_\* in the c file. 
Or is it possible to typedef the bt_mesh_cfg_mod_pub to bt_mesh_cli_mod_pub.